client/modules/accounts/templates/signUp/signUp.js

```diff
@@ -29,44 +30,69 @@ Template.loginFormSignUpView.events({
    */
   "submit form": function (event, template) {
     event.preventDefault();
-
-    // var usernameInput = template.$(".login-input--username");
+    const usernameInput = template.$(".login-input-username");
     const emailInput = template.$(".login-input-email");
     const passwordInput = template.$(".login-input-password");
-
+    const username = usernameInput.val().trim();
     const email = emailInput.val().trim();
     const password = passwordInput.val().trim();
-
+    const validatedUsername = LoginFormValidation.username(username);
     const validatedEmail = LoginFormValidation.email(email);
     const validatedPassword = LoginFormValidation.password(password);
-
     const templateInstance = Template.instance();
     const errors = {};
-
     templateInstance.formMessages.set({});
-
+    if (validatedUsername !== true) {
+      errors.email = validatedUsername;
+    }
     if (validatedEmail !== true) {
       errors.email = validatedEmail;
     }
-
     if (validatedPassword !== true) {
       errors.password = validatedPassword;
     }
+    let vendorDetails = {};
+    if (Session.get("signupas") === "vendor") {
+      shopName = template.$(".shop-name").val().trim();
+      shopPhone = template.$(".shop-phone").val().trim();
+      shopAddress = template.$(".shop-address").val().trim();
+
+      const validatedShopName = LoginFormValidation.shopName(shopName);
+      const validatedShopPhone = LoginFormValidation.shopPhone(shopPhone);
+      const validatedShopAddress = LoginFormValidation.shopAddress(shopAddress);
+      if (validatedShopName !== true) {
+        errors.shopName = validatedShopName;
+      }
+      if (validatedShopPhone !== true) {
+        errors.shopPhone = validatedShopPhone;
+      }
+      if (validatedShopAddress !== true) {
+        errors.shopAddress = validatedShopAddress;
+      }
 
+      vendorDetails = {
+        vendorDetails: [{
+          shopName: shopName,
+          shopPhone: shopPhone,
+          shopAddress: shopAddress,
+          isVendor: true,
+          shopActive: false
+        }]
+      };
+    }
     if ($.isEmptyObject(errors) === false) {
       templateInstance.formMessages.set({
         errors: errors
       });
       // prevent signup
       return;
     }
-
     const newUserData = {
-      // username: username,
+      username: username,
```

Comment:
I'm curious if the entire codebase support ES6. If it does, when the key and value ar the same, you only need the property.

E.G:
```javascript
    const newUserData = {
       email, 
       password, 
       // etc
     }
```